### PR TITLE
Add more dart versions

### PR DESCRIPTION
**Pull request by miDeb**
Infra PR: https://github.com/compiler-explorer/infra/pull/895

Adds dart version 2.18.5 (I think it makes sense not to include all patch versions) and the latest dev channel build (currently version 3.0).

### CODE REVIEW

File: etc/config/dart.amazon.properties [hunk 1]
```diff
@@ -5,11 +5,19 @@ supportsExecute=true
 compilerType=dart
 objdumper=/opt/compiler-explorer/gcc-11.1.0/bin/objdump
 
-group.dart.compilers=dart2171:dart2161:dart2151:dart2144:dart2134:dart2124:dart2105:dart293:dart284
+group.dart.compilers=dev:dart2185:dart2171:dart2161:dart2151:dart2144:dart2134:dart2124:dart2105:dart293:dart284
```
Comment (RubenRBS) [inline etc/config/dart.amazon.properties:8]: Can you please change `dev` to `dartdev`? Compiler ids are global and having one called `dev` is not ideal

**Review by RubenRBS — CHANGES_REQUESTED** [review]
👍 Thanks!

File: etc/config/dart.amazon.properties (hunk 1 above)
Comment (miDeb) [reply etc/config/dart.amazon.properties]: Thanks, done. I didn't realize they those were global ids. I guess in that case you may also want to rename the rust variants that are only called "nightly" and "beta".

Comment (RubenRBS) [reply etc/config/dart.amazon.properties]: Oh nice catch! I didn't know those versions had that id.

Unfortunately IDs are inmutable so we can't change those once they go live, so we're stuck with those bad apples until the end of times, yay!

**Comment by RubenRBS** [comment]
That was quick, thank you! I'll merge now and this should go live in the next few days :)

**Pull request merged**